src/InlineInput/NumberInlineInput.tsx

```diff
@@ -7,63 +7,72 @@
 import React, { FC } from 'react';
 
 import { isFactor } from '../Slider/factor';
-import { RangeProp } from '../Slider/rangeShape';
+import {
+    getStep,
+    isValues,
+    Range,
+    RangeOrValues,
+    useValidatedRangeOrValues,
+    Values,
+} from '../Slider/range';
 import InlineInput from './InlineInput';
 
 import './number-inline-input.scss';
 
-const isInRange = (
-    value: number,
-    { min, max, decimals = 0, step = 0, explicitRange = [] }: RangeProp
-) =>
+export interface Props {
+    disabled?: boolean;
+    value: number;
+    range: RangeOrValues;
+    onChange: (value: number) => void;
+    onChangeComplete?: (value: number) => void;
+}
+
+const isInValues = (value: number, values: Values) => values.includes(value);
+
+const isInRange = (value: number, { min, max, decimals, step }: Range) =>
     value >= min &&
     value <= max &&
     value === Number(value.toFixed(decimals)) &&
-    (step > 0 ? isFactor(value, step) : true) &&
-    (explicitRange.length > 0 ? explicitRange.indexOf(value) !== -1 : true);
+    (step == null ? true : isFactor(value, step));
```

Comment:
to fix the issue for `{min: 1, max: 7, step: 2}` in 'NumberInlineInput' we need to do the below here 

```suggestion
    (step == null ? true : isFactor(value - min, step));
```

This shift the range to make use of isFactor